libs/forms/src/lib/components/form-generator/button/button.component.ts

```diff
@@ -2,36 +2,38 @@ import {
   ChangeDetectionStrategy,
   Component,
   Input,
-  OnChanges,
-  OnInit,
-  SimpleChanges,
+  computed,
 } from '@angular/core';
-import { FormGroup } from '@angular/forms';
-import { BehaviorSubject, Observable } from 'rxjs';
+import { toObservable } from '@angular/core/rxjs-interop';
+import { mergeMap } from 'rxjs';
 import { FormsLib } from '../../../forms.model';
 import { is } from '../../../utils';
 import { dynamicPropertyEvaluation$ } from '../../../utils/dynamic-property-evaluation.util';
+import { FormGeneratorBaseComponent } from '../form-generator.base';
 
 @Component({
   selector: 'lib-button',
   templateUrl: './button.component.html',
   styleUrls: ['./button.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ButtonComponent implements OnInit, OnChanges {
+export class ButtonComponent extends FormGeneratorBaseComponent {
   @Input() button?: FormsLib.Button | null = null;
-  @Input() formGroup = new FormGroup({});
 
-  /** Datafields for dynamic data */
-  @Input() datafields?: FormsLib.Datafields | null = {};
-
-  public visible$: Observable<boolean> = new BehaviorSubject(true);
+  /** Dynamically determine visibility */
+  public visible$ = toObservable(
+    computed(() => ({ formGroup: this.formGroup, button: this.button }))
+  ).pipe(
+    mergeMap(({ formGroup, button }) =>
+      dynamicPropertyEvaluation$(button?.visible, formGroup)
+    )
+  );
 
   public is = is;
 
-  constructor() {}
-
-  ngOnInit(): void {}
+  constructor() {
+    super();
```

Comment:
isn't this the same as not having a constructor? (Looks like this pattern is used a lot)

Reply:
Super is required for extended classes to manage dependency injection. Even if you don't have any dependencies it's still required and will throw a TS error if you don't have it.

> export class HtmlComponent extends FormGeneratorBaseComponent {